corehq/apps/locations/models.py

```diff
@@ -566,6 +568,7 @@ def unarchive(self):
         from corehq.apps.users.models import CommCareUser
         for loc in itertools.chain(self.get_descendants(include_self=True), self.get_ancestors()):
             loc.is_archived = False
+            loc.archived_at = None
```

Comment:
I thought it would be confusing if archived_at is set on an unarchived location, so reset it.